pkg/ibm/ibmp.go

```diff
@@ -140,7 +119,55 @@ func (r IBMPowerDynamicConfig) GetInstanceAddress(kubeClient client.Client, ctx
 }
 
 func (r IBMPowerDynamicConfig) ListInstances(kubeClient client.Client, ctx context.Context, instanceTag string) ([]cloud.CloudVMInstance, error) {
-	return nil, fmt.Errorf("not impelemented")
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("Listing instances by tag:" + instanceTag)
+	service, err := r.authenticate(kubeClient, ctx)
+	if err != nil {
+		return nil, err
+	}
+	builder := core.NewRequestBuilder(core.GET).WithContext(ctx)
+	builder.EnableGzipCompression = service.GetEnableGzipCompression()
+
+	pathParamsMap := map[string]string{
+		"cloud": r.pCloudId(),
+	}
+	_, err = builder.ResolveRequestURL(r.Url, `/pcloud/v1/cloud-instances/{cloud}/pvm-instances`, pathParamsMap)
+	if err != nil {
+		return nil, err
+	}
+	builder.AddHeader("CRN", r.CRN)
+	builder.AddHeader("Accept", "application/json")
+
+	request, err := builder.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	instances := models.PVMInstances{}
+	_, err = service.Request(request, &instances)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make([]cloud.CloudVMInstance, 0, len(instances.PvmInstances))
+	for _, instance := range instances.PvmInstances {
+		if !strings.HasPrefix(*instance.ServerName, instanceTag) {
+			continue
+		}
+		identifier := cloud.InstanceIdentifier(*instance.PvmInstanceID)
+		createdAt, err := time.Parse(time.RFC3339Nano, instance.CreationDate.String())
```

Comment:
isn't CreationDate already a time value? If we need to do this parsing here, that models library is not doing much for us at all...

Reply:
yep, its an wrapper wich can be just cast to `Time`, fixed